api/resource/admin.py

```diff
@@ -134,29 +133,24 @@ def admin_update(email: str) -> Tuple[Dict[str, str], int]:
           500 - otherwise
         """
         body = validate_request_body(AdminUpdateSchema, request.json)
+        print(body)
         if isinstance(body, str):
             return {"msg": body}, 400
 
-        admin = db.admins.find_one({"email": body["email"]})
-
+        admin = db.admins.find_one({"email": email})
         if "password" in body:
             if body["password"] != body["password_confirmation"]:
                 return (
                     {"msg": "`password` and `password_confirmation` don't match"},
                     401,
                 )
             del body["password_confirmation"]
-            body["password"] = generate_password_hash(body["password"])
-        else:
-            body["password"] = admin["password"]
```

Comment:
Still need this part as well, this is the part that actually creates the new password

Reply:
Generating the password_hash on L148